freqtrade/worker.py

```diff
@@ -86,7 +86,7 @@ def _worker(self, old_state: Optional[State]) -> State:
         # Log state transition
         if state != old_state:
 
-            if old_state != State.RELOAD_CONFIG:
+            if state != State.RELOAD_CONFIG:
```

Comment:
I'm not sure i understand the point of this (what this tries to accomplish).

What this changes is when this is executed (before/after the reload). 
Please explain which messages / in which scenario you think this makes a difference.